java-frontend/src/main/java/org/sonar/java/resolve/Resolve.java

```diff
@@ -474,30 +475,37 @@ private Resolution findMethodByStrictThenLooseInvocation(Env env, JavaType callS
 
   private Resolution findMethod(Env env, JavaType callSite, JavaType site, String name, List<JavaType> argTypes, List<JavaType> typeParams,
                                 boolean looseInvocation, boolean varArity) {
-    JavaType superclass = site.getSuperType();
-    Resolution bestSoFar = unresolved();
+    return findMethod(env, callSite, site, name, argTypes, typeParams, looseInvocation, varArity, new HashSet<>());
+  }
+
+  private Resolution findMethod(Env env, JavaType callSite, JavaType site, String name, List<JavaType> argTypes, List<JavaType> typeParams,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Method has 9 parameters, which is greater than 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00107)